lib/mpl_toolkits/mplot3d/tests/test_axes3d.py

```diff
@@ -2124,3 +2124,52 @@ def test_panecolor_rcparams():
                          'axes3d.zaxis.panecolor': 'b'}):
         fig = plt.figure(figsize=(1, 1))
         fig.add_subplot(projection='3d')
+
+
+# test cases came from GH#8990
+@check_figures_equal(extensions=["png"])
+def test_mutating_input_arrays_y_and_z(fig_test, fig_ref):
+    ax1 = fig_test.add_subplot(111, projection='3d')
+    x = [1, 2, 3]
+    y = [0.0, 0.0, 0.0]
+    z = [0.0, 0.0, 0.0]
+    ax1.plot(x, y, z, 'o-')
+
+    ax1.set_ylim([0, 4])
+    ax1.set_zlim([0, 4])
+    fig_test.draw_without_rendering()
+
+    # mutate y,z to get a nontrivial line
+    y[:] = [1, 2, 3]
+    z[:] = [1, 2, 3]
+    fig_test.draw_without_rendering()
+
+    # draw the same plot without mutating x and y
+    ax2 = fig_ref.add_subplot(111, projection='3d')
+    x = [1, 2, 3]
+    y = [0.0, 0.0, 0.0]
+    z = [0.0, 0.0, 0.0]
+    ax2.plot(x, y, z, 'o-')
+
+    ax2.set_ylim([0, 4])
+    ax2.set_zlim([0, 4])
+
+
+# test cases came from GH#8990
+@check_figures_equal(extensions=["png"])
+def test_mutating_input_arrays_sin_3d(fig_test, fig_ref):
```

Comment:
What is this testing that the other test is not testing?

Reply:
It's testing that the second curve doesn't affect the first curve as reported [here. ](https://github.com/matplotlib/matplotlib/issues/8990#issuecomment-1131892593)

Reply:
In that example the important change is mutating the input not that we plotted it a second time (if you go back to 3.5.2 and remove the second plot I expect it to be broken).

Reply:
Ok. In that case, should we remove the second test entirely?